### DISCUSSION TITLE
Packet Structure

### DISCUSSION THREAD

**Discussion by georgev93** [emailing-files-is-still-better/Ir-Blinky#23] [General]
We initially thought the message was `preamble + data`. That turned out not to be the case, in that there was no consistent start of the packet from address to address.

**Comment by georgev93** [comment]
Could be `data + checksum`

**Comment by georgev93** [comment]
As of right now, there is one pattern I see as consistent between the two messages, the first 10 bits is equal to the second 10 bits. Coincidence? Perhaps we could do a brute force of the 20 lowest bits, then copy the 10 bits in the front to the first 10 bits. That would narrow our scope to only brute forcing 20 bits

**Reply by georgev93** [reply]
I wrote the `setDataPattern` function in a way that should make this easy. If you want to set the lower 20 bits, you could do that with

``` C
uint32_t nextPattern = 0b01010101011010101010;

setDataPattern(nextPattern, 20, &nextArrayStartPtr[10]);
```

It just places the pattern starting after the 10th bit (arrays are indexed from 0, so [10] points at the 11th bit).

Then you could always copy with memcpy, or something really simple like:

```C
for(int i=0; i<10; i++) {
  nextArrayStartPtr[i] = nextArrayStartPtr[i+10];
}
```

**Reply by emailing-files-is-still-better** [reply]
You're right. I did not see this at all. I'm also not convinced I would put this in the correct place either lol

**Comment by emailing-files-is-still-better** [comment]
Yeah I think that would help. You know what would be crazy. If the next time the first 10 matched the second 10 was #95 when incrementing up from address #92.
